agent/engine/task_manager_test.go

```diff
@@ -21,6 +21,7 @@ import (
 	"time"
 
```

Comment:
There's a lot of changed tests -- but no new tests. You need unit and integration tests for this.